py/compile.c

```diff
@@ -3112,6 +3118,9 @@ STATIC void compile_scope(compiler_t *comp, scope_t *scope, pass_kind_t pass) {
             scope->num_pos_args = 1;
         }
 
+#if MICROPY_PY_SYS_SETTRACE
+        EMIT_ARG(set_source_line, pns->source_line);
```

Comment:
This one is useful in general (not just for settrace) because it'll give much better line number reporting in comprehensions.  Currently we get "line 1" for comprehensions.

Reply:
Should I just remove the `#if` and keep it in this PR? Or separate PR?

Reply:
I know it's just one line but I think it's a good candidate for a separate commit.  I'm happy to do the commit, or feel free to make a separate PR.